pytorch_pfn_extras/training/extensions/_snapshot.py

```diff
@@ -349,6 +349,9 @@ def initialize(  # type: ignore[override]
             # manager.
             assert writer is not None
             loaded_fn = _find_latest_snapshot(self.filename, outdir, writer.fs)
+            if not loaded_fn:
+                # Try searching in ``writer.out_dir`` if there is no snapshot in ``outdir``
+                loaded_fn = _find_latest_snapshot(self.filename, writer.out_dir, writer.fs)
```

Comment:
```suggestion
            loaded_fn = _find_latest_snapshot(self.filename, writer.out_dir, writer.fs)
            if not loaded_fn:
                loaded_fn = _find_latest_snapshot(self.filename, outdir, writer.fs)
```

When a snapshot object has a writer object, I want the value of the out_dir property of the writer object to be used first, so I want the order to be switched.
When the snapshot object does not have a writer object, the value of `outdir` matches `manager.writer.out_dir`, so I think it is ok to change as suggested.